internal/image/resource_image.go

```diff
@@ -444,6 +462,168 @@ func (r ImageResource) SyncState(ctx context.Context, tfState *tfsdk.State, serv
 	return tfState.Set(ctx, &m)
 }
 
+func (r ImageResource) createImageFromSourceFile(ctx context.Context, resp *resource.CreateResponse, plan *ImageModel) {
+	var sourceFileModel SourceFileModel
+
+	diags := plan.SourceFile.As(ctx, &sourceFileModel, basetypes.ObjectAsOptions{})
+	if diags.HasError() {
+		resp.Diagnostics.Append(diags...)
+		return
+	}
+
+	remote := plan.Remote.ValueString()
+	project := plan.Project.ValueString()
+	server, err := r.provider.InstanceServer(remote, project, "")
+	if err != nil {
+		resp.Diagnostics.Append(errors.NewInstanceServerError(err))
+		return
+	}
+
+	var dataPath, metadataPath string
+	if sourceFileModel.MetadataPath.IsNull() {
+		// Unified image
+		metadataPath = sourceFileModel.DataPath.ValueString()
+	} else {
+		dataPath = sourceFileModel.DataPath.ValueString()
+		metadataPath = sourceFileModel.MetadataPath.ValueString()
+	}
+
+	var image api.ImagesPost
+	var createArgs *incus.ImageCreateArgs
+
+	imageType := "container"
+	// TODO: should https urls also be supported?
```

Comment:
Currently, there is no acceptance test for the code path using an URL for the imported image.

Reply:
Discussed with @stgraber, we are fine with not covering this code path with acceptance tests.